### PR TITLE
docs: header propagation content moves from overview into dedicated page

### PR DESCRIPTION
**Pull request by abernix**
The existing page already was in place, this just moves the content from the
Overview to the dedicated home.

### CODE REVIEW

File: docs/source/configuration/header-propagation.mdx [hunk 1]
```diff
@@ -6,4 +6,110 @@ description: Configuring the HTTP headers which are sent to subgraphs
 
 import { Link } from 'gatsby';
 
-TODO
\ No newline at end of file
+You can customize which HTTP headers the Apollo Router includes in its requests to your individual subgraphs. You specify these headers on a per-subgraph basis in your [YAML configuration file](#configuration-file).
+
+More specifically, each the headers plugin include can be configured with an array of header-related actions for the router to perform (in order) before sending a request:
```
Comment (StephenBarlow) [inline docs/source/configuration/header-propagation.mdx:11]: I think there's some garbled language in the first half of this sentence

File: docs/source/configuration/header-propagation.mdx [hunk 2]
```diff
@@ -6,4 +6,110 @@ description: Configuring the HTTP headers which are sent to subgraphs
...
+
+```yaml
+- insert:
+    name: "sent-from-our-apollo-router"
+    value: "indeed"
+```
+
+### Example
+
+Here's an example of customizing different headers for two subgraphs: `accounts` and `products`. These two subgraphs are both sent `router-subgraph-name` headers with either the `accounts` or `products` value, respectively. Any other subgraph receives _no_ headers.
```
Comment (StephenBarlow) [inline docs/source/configuration/header-propagation.mdx:88]: This clarifying sentence doesn't reflect the configuration below, I don't think? Should reframe to match

File: docs/source/configuration/header-propagation.mdx [hunk 3]
```diff
@@ -6,4 +6,110 @@ description: Configuring the HTTP headers which are sent to subgraphs
...
+    - propagate:
+        named: "some-header"
+        default: "default-value"
+        rename: "destination-header"
+    # All subgraphs will remove the "x-legacy-account-id" header.
+    - remove:
+        named: "x-legacy-account-id"
+    - remove:
+        matching: "x-legacy-account-id"
+    # Subgraph specific configuration
```
Comment (StephenBarlow) [inline docs/source/configuration/header-propagation.mdx:108]: "Subgraph-specific"

**Review by StephenBarlow — APPROVED** [review]
Looks good with suggested edits!

**Comment by abernix** [comment]
@BrynCooke Do you mind taking a look at this and addressing @StephenBarlow's feedback above, in addition to making sure this aligns with the technical implementation?

**Comment by BrynCooke** [comment]
I updated the examples and trimmed down the text a bit. @StephenBarlow probably worth you taking another pass at this.

**Review by StephenBarlow — APPROVED** [review]
Looks good to me!

**Pull request merged**